feature: add Thai language support

iakgoog opened the pull request:
Adds comprehensive Thai language support to Pororoca following the existing multi-language pattern. #181 

  Changes Made

  Core Language System:
  - Added Thai enum to Language.cs with th-th LCID mapping

  Localization:
  - Created complete Thai translation file (th-th.i18n_lang.json) with 370+ strings
  - Updated i18n_keys.json with Thai language menu option

  Documentation:
  - Added Thai README translation (README_th.md)
  - Updated all 8 existing README files with Thai language links

  UI Integration:
  - Added Thai language menu item to MainWindow.xaml
  - Updated MainWindowViewModel.cs with Thai language properties and commands

  Windows Installer:
  - Added Thai language file loading to Installer.nsi
  - Added complete Thai translations for all 17 installer strings
  - Updated NsisMultiUserLang.nsh with 27 Thai installer-specific strings
  - Added Thai to language selection dialog

iakgoog committed:
feature: add Thai language support

alexandrehtrb commented:
@iakgoog, thanks for you contribution.
This PR looks like has been made with the help of AI code.
I kindly ask you to manually check Thai generated texts, please check if they are in good quality. I am leaving this PR open because your GitHub profile says that you are from Thailand, so I trusting you in good faith.
I always check translations, but I can't determine if the texts are close to what a native speaker would say / write / read.

iakgoog commented:
@alexandrehtrb definitely, I would carefully check every translation and compare to the meaning on the context that the translation appear

iakgoog commented:
Hi @alexandrehtrb 
I've carefully checked and improve some translation.
Some terms are kept in English because they are commonly known among Thai developers and convey their meaning more clearly than their Thai translations (especially in programming context)

alexandrehtrb commented on src/Pororoca.Desktop.WindowsInstaller/Include/NsisMultiUserLang.nsh:
```diff
@@ -1392,6 +1392,37 @@
 	LangString MULTIUSER_ELEVATION_ERROR ${LANG_POLISH} "Nie można podnieść poziomu uprawnień. Błąd: {ERROR}."
 !endif
 
+!ifdef LANG_THAI
```
`LANG_THAI` is already defined below `LANG_TATAR`, it just needs to replace the strings from English to Thai

iakgoog replied:
fixed

alexandrehtrb commented on src/Pororoca.Desktop/Localization/i18n_keys.json:
```diff
@@ -281,6 +281,7 @@
   "TopMenuLanguage/Russian",
   "TopMenuLanguage/SimplifiedChinese",
   "TopMenuLanguage/Spanish",
+  "TopMenuLanguage/Thai",
```
You forgot translation for this key on file `en-gb.i18n_lang.json`

iakgoog replied:
fixed

alexandrehtrb commented on src/Pororoca.Desktop/Localization/th-th.i18n_lang.json:
```diff
@@ -0,0 +1,372 @@
+  "Collection/AddHttpRequest": "เพิ่มคำขอ HTTP",
+  "Collection/AddWebSocketConnection": "เพิ่ม WebSocket",
+  "Collection/ChooseImportExportFormat": "เลือกรูปแบบ",
+  "Collection/ExportCollection": "ส่งออกคอลเล็กชัน...",
```
Replace occurrences of `...` for `…` (1 char)

iakgoog replied:
fixed

iakgoog committed:
feat: improve Thai language support

- Add Thai to language selection menu
- Reorganize Thai installer strings in NSIS file
- Replace triple dots with proper ellipsis character in Thai translations
